server/eventlog.go

```diff
@@ -135,11 +141,22 @@ func getVerifiedCosState(coscel cel.CEL, pcrs *tpmpb.PCRs) (*pb.AttestedCosState
 			return nil, err
 		}
 
+		// TODO: Add support for post-separator container data
+		if seenSeparator {
+			return nil, fmt.Errorf("found COS Event Type %v after LaunchSeparator event", cosTlv.EventType)
+		}
+
 		switch cosTlv.EventType {
 		case cel.ImageRefType:
+			if cosState.Container.GetImageReference() != "" {
+				return nil, fmt.Errorf("found duplicate ImageRef event")
```

Comment:
nit: "found more than one ImageRef event". Found duplicate makes it sound like everything in the event is the same (including the ref string).

Same below with ImageDigest and ID

Reply:
Good catch agreed.